File: backend/Services/ApiGateway/ApiGateway/services/AuthService.cs
using Microsoft.AspNetCore.Identity.Data;
using Shared.Contracts.DTOs.Auth.Responses;
using LoginRequest = Shared.Contracts.DTOs.Auth.Requests.LoginRequest;
using RegisterRequest = Shared.Contracts.DTOs.Auth.Requests.RegisterRequest;

namespace ApiGateway.services;

public class AuthService(IHttpClientFactory clientFactory)
{
    private readonly HttpClient _httpClient =
        clientFactory.CreateClient("AuthService");

    public async Task<string> GetTestAsync()
    {
        var response = await _httpClient.GetAsync("api/test");

        if (response.IsSuccessStatusCode)
        {
            var result = await response.Content.ReadAsStringAsync();

            return result;
        }

        return null;
    }
    
    public async Task<LoginResponse> LoginAsync(LoginRequest request)
    {
        var response = await _httpClient
            .PostAsJsonAsync("api/auth/login", request);

        if (response.IsSuccessStatusCode)
        {
            var result = await response.Content
                .ReadFromJsonAsync<LoginResponse>();

            return result;
        }

        return null;
    }
    
    public async Task<RegisterResponse> RegisterAsync(RegisterRequest request)
    {
        var response = await _httpClient
            .PostAsJsonAsync("api/auth/register", request);

        if (response.IsSuccessStatusCode)
        {
            var result = await response.Content
                .ReadFromJsonAsync<RegisterResponse>();

            return result;
        }

        return null;
    }
}

File: frontend/Frontend/neurosync/src/services/auth-service/authService.ts
import {apiService} from "../api-service/apiService.ts";
import type {LoginRequest} from "../../models/auth/login/requests/LoginRequest.ts";
import type {LoginResponse} from "../../models/auth/login/responses/LoginResponse.ts";
import type {RegisterRequest} from "../../models/auth/register/requests/RegisterRequest.ts";
import type {RegisterResponse} from "../../models/auth/register/responses/RegisterResponse.ts";

export const authService = {
    login: (data: LoginRequest) =>
        apiService.post<LoginResponse>("login", data),

    register: (data: RegisterRequest) =>
        apiService.post<RegisterResponse>("register", data),
}